Platform/Plugins/com.tle.platform.common/src/com/tle/common/util/ExecUtils.java

```diff
@@ -52,6 +54,125 @@
+    if (pid < 1) {
+      throw new IllegalArgumentException("Process ID should be greater than 0");
+    }
+    // get child process PIDs and kill them
+    getChildUnixProcessPids(pid).ifPresent(ExecUtils::sendSigKill);
+
+    // kill process itself, after all the child processes have been terminated
+    sendSigKill(pid);
+  }
+
+  /**
+   * Runs pgrep -P for a given Process ID, to get a list of child processes as Process IDs.
+   *
+   * @param pid The parent process ID to check for child processes.
+   * @return Optional<String[]> An array of process IDs for the children of pid.
+   */
+  public static Optional<int[]> getChildUnixProcessPids(int pid) {
+    Optional<int[]> pids = Optional.empty();
+    try {
+      Process getChildPid = Runtime.getRuntime().exec("pgrep -P " + pid);
+      getChildPid.waitFor();
+      StringBuilder childPid = new StringBuilder();
+      if (getChildPid.exitValue() == 0) {
+        CharStreams.copy(new InputStreamReader(getChildPid.getInputStream()), childPid);
+      } else {
+        StringBuilder errorOutput = new StringBuilder();
+        CharStreams.copy(new InputStreamReader(getChildPid.getErrorStream()), errorOutput);
+        LOGGER.debug("getChildPid function did not run properly.\n" + errorOutput);
+      }
+      getChildPid.destroy();
+      // convert string to array of ints
+      pids =
+          Optional.of(
+              Arrays.stream(childPid.toString().replaceAll("\n", " ").split(" "))
+                  .mapToInt(Integer::parseInt)
+                  .toArray());
+    } catch (IOException | InterruptedException e) {
+      LOGGER.error("Error getting child processes for: " + pid, e);
+    } catch (NumberFormatException e) {
+      LOGGER.error("Output of getChildPids command not parsable as integers", e);
+    }
+    return pids;
+  }
+
+  /**
+   * Creates a process which then sends a SIGKILL signal to an array of given processes.
+   *
+   * @param pids the Process IDs of the processes to kill.
+   * @return the exitValue of the SIGKILL process (not the process being killed)
+   */
+  public static int sendSigKill(int[] pids) {
+    try {
+      StringBuilder command = new StringBuilder("kill -9 ");
+      for (int pid : pids) {
+        command.append(pid).append(" ");
+      }
+      LOGGER.debug("Running command: " + command);
+      Process sigKill = Runtime.getRuntime().exec(String.valueOf(command));
+      sigKill.waitFor();
+      int returnValue = sigKill.exitValue();
+      if (sigKill.exitValue() == 0) {
+        StringBuilder successOutput = new StringBuilder();
+        CharStreams.copy(new InputStreamReader(sigKill.getInputStream()), successOutput);
+        LOGGER.debug("Output of kill function: " + successOutput);
+      } else {
+        StringBuilder errorOutput = new StringBuilder();
+        CharStreams.copy(new InputStreamReader(sigKill.getErrorStream()), errorOutput);
+        LOGGER.debug("kill function did not run properly.\n" + errorOutput);
```

Comment:
Be useful for administrators to know when this kind of thing is not functioning correctly.

```suggestion
        LOGGER.warn("Attempt to terminate processes failed (" + command + "):\n" + errorOutput);
```